memtable/wbwi_memtable.h

```diff
@@ -13,21 +13,29 @@ namespace ROCKSDB_NAMESPACE {
 // a transaction (which is based on WBWI) into the DB as an immutable memtable.
 //
 // REQUIRE overwrite_key to be true for the WBWI
-// Since the keys in WBWI do not have sequence number, the memtable needs to be
+// Since the keys in WBWI do not have sequence number, this class is responsible
+// for assigning sequence numbers to the keys. This memtable needs to be
 // assigned a range of sequence numbers through AssignSequenceNumbers(seqno)
 // before being available for reads.
-// With overwrite_key = true, WBWI keeps track of the most recent update for
-// each key, and each such key will be assigned seqno.upper_bound during reads.
-// One exception is during flush, consider the following scenario:
+//
+// The sequence number assignment uses the update count for each key
+// tracked in WBWI (see WBWIIterator::GetUpdateCount()). For each key, the
+// sequence number assigned is seqno.lower_bound + update_count - 1. So more
+// recent updates will have higher sequence number.
+//
+// WBWI with overwrite mode keeps track of the most recent update for each key,
+// so this memtable contains one update per key usually. However, there is a
+// special case where this memtable needs to emit an extra SingleDelete even
```

Comment:
I assume discussion of special case for merge comes later?

Reply:
Yes, it will come later. It should be a small change where for merge we don't track only the most recent update.